### ISSUE TITLE
aggregate() continues even though get_ifcfile() returns None

### ISSUE THREAD

**Issue by mrevl** [yorikvanhavre/FreeCAD-NativeIFC#66]
Use case:
I'm using native ifc along a regular fcstd. Mostly for exporting to ifc, hopefully in the future for importing non-freecad generated elements back into freecad. 

Having a project does not guarantee having an existing ifcfile.
Context: ifc_tools.py

def aggregate(obj, parent):
    """Takes any FreeCAD object and aggregates it to an existing IFC object"""

    proj = get_project(parent)
    if not proj:
        FreeCAD.Console.PrintError("The parent object is not part of an IFC project\n")
        return
    ifcfile = get_ifcfile(proj)

....4 calls down the line....
get_export_preferences()
prefs["SCHEMA"] = ifcfile.wrapped_data.schema_name()
AttributeError: 'NoneType' object has no attribute 'wrapped_data' as ifcfile was None all along

**Comment by mrevl** [comment]
Please note that I can reproduce this if I select multiple objects and then try to do a bim preflight check on selected items.
_However_ I can not reproduce this is if I select objects individually.
So far I can determine that this is most likely restricted to window/door openings. Once I have more time I'll try to dig further.

This is FreeCAD 0.22.0, Libs: 0.22.0devR37841 (Git)  (weekly flatpak image as of Jun 22th 2024)

Traceback (most recent call last):
  File "/tmp/.mount_freecaFWrYAi/usr/Mod/BIM/nativeifc/ifc_observer.py", line 200, in convert
    newobj = ifc_tools.aggregate(obj, doc)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/.mount_freecaFWrYAi/usr/Mod/BIM/nativeifc/ifc_tools.py", line 963, in aggregate
    product = create_product(obj, parent, ifcfile)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/.mount_freecaFWrYAi/usr/Mod/BIM/nativeifc/ifc_tools.py", line 1013, in create_product
    representation, placement = create_representation(obj, ifcfile)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/.mount_freecaFWrYAi/usr/Mod/BIM/nativeifc/ifc_tools.py", line 1049, in create_representation
    prefs, context = get_export_preferences(ifcfile)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/.mount_freecaFWrYAi/usr/Mod/BIM/nativeifc/ifc_tools.py", line 1076, in get_export_preferences
    prefs["SCHEMA"] = ifcfile.wrapped_data.schema_name()
                      ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'wrapped_data'

**Comment by mrevl** [comment]
Workaround found!

After a bit of digging I discovered that my "document" had IFC attributes. I suspect that this is from an earlier IFC conversion where attributes got added to the wrong object. Tools -> document utility is an easy way to check this, first unpack the freecad file into a directory and then check document.xml / compare it to a brand new document. 

My workaround was to create a new document, via BIM -> Manage project and then to import the faulty document. This allowed me to keep everything as is with a new, valid "document" root. 

Closing this as this is a bug on a situation that shouldn't have existed in the first place.
